### ISSUE TITLE
HiddenMarkovModel.freeze_distribution() fails 

### ISSUE THREAD

**Issue by shpigi** [jmschrei/pomegranate#8]
```python
    import pomegranate as pm
    model=pm.HiddenMarkovModel()
    model.add_state(pm.State(pm.NormalDistribution(0,1),'s1'))
    model.freeze_distributions()

In [5]: model.freeze_distributions()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-29-49df663003f8> in <module>()
----> 1 model.freeze_distributions()

pomegranate/hmm.pyx in pomegranate.hmm.HiddenMarkovModel.freeze_distributions (pomegranate/hmm.c:5076)()

TypeError: 'NoneType' object is not iterable
```

The same happens if I `bake()` the model first. I guess it reached the start or end state and tries to freeze its `None` distribution

**Comment by jmschrei** [comment]
Yeah, looks like it'll fail on any silent state. I'm going to push an update soon which will fix this.

**Comment by jmschrei** [comment]
Fixed in v0.2.3.
